src/main/java/org/osc/controller/nsc/utils/RedirectionApiUtils.java

```diff
@@ -168,18 +168,45 @@ public NetworkElementEntity txNetworkElementEntityById(Long id) {
         return this.em.find(NetworkElementEntity.class, id);
     }
 
-    public void removeSingleInspectionHook(InspectionHookEntity inspectionHookEntity) {
+    public void removeSingleInspectionHook(String hookId) {
```

Comment:
not sure why we have so many tranasactions for this one method...why cant this all be done in one transaction?